sass/_header.scss

```
@@ -207,3 +187,50 @@
         }
     }
 }
+
+.submenu-checkbox {
+    display: none;
+
+    @media screen and (max-width: 1023px) {
+        &:not(:checked) + .submenu-title-container + .section-submenu-wrap {
```

Given you moved the different submenu checkboxes into distinct parent containers (`section-wrap`), you can shorten this to

```suggestion
        &:not(:checked) ~ .section-submenu-wrap {
```